File: app/getconf.py
import json
import logging
import sys
import time
from io import StringIO
from ipaddress import ip_address
from os.path import exists


from tcp_client import tcp_client

ip='192.168.0.17'
if len(sys.argv) == 1:
    ip = sys.argv[1]


device_buf = StringIO()

a = tcp_client(ip, timeout=0.1)
a._initSocket()
if a._connect:
        device_info = a._device_info()
        device_info_str = f'  - ip: {ip}\n'
        device_info_str += f'    did: {a._device_id}\n'
        device_info_str += f'    pid: {a._pid}\n'
        device_info_str += f'    dmn: {a._device_model_name}\n'
        device_info_str += f'    dpid: {a._dpid}\n'
        device_info_str += f'    device_type: {a._device_type_code}\n'

        print(device_info_str)
        print(device_info)

        state = a.query()
        print(a.query())

File: app/getconf.js
import { TcpClient } from './tcp_client.js';

// Récupération de l'IP depuis les arguments
let ip = '192.168.0.17';
if (process.argv.length > 2) {
  ip = process.argv[2];
}

console.log(`Scanning IP: ${ip}`);

async function main() {
  const client = new TcpClient(ip, 100);
  
  const connected = await client._initSocket();
  
  if (connected) {
    const device_info = await client._device_info();
    
    const device_info_str = `  - ip: ${ip}
    did: ${client._device_id}
    pid: ${client._pid}
    dmn: ${client._device_model_name}
    dpid: ${JSON.stringify(client._dpid)}
    device_type: ${client._device_type_code}`;
    
    console.log(device_info_str);
    console.log('\nDevice Info:', device_info);
    
    const state = await client.query();
    console.log('\nDevice State:', state);
    
    client.disconnect();
  } else {
    console.log('Failed to connect to device');
  }
}

main().catch(err => {
  console.error('Error:', err.message);
  process.exit(1);
});
